src/App.tsx

```diff
@@ -1,6 +1,34 @@
+import { useState } from "react";
 import styles from "./App.module.css";
+import Frame1 from "./components/frame1.tsx";
+import Frame2 from "./components/frame2.tsx";
 
 function App() {
+  const [frame, setFrame] = useState(0);
```

Comment:
```suggestion
  const [frame, setFrame] = useState<0 | 1>(0);
```

frame은 0 또는 1만 들어갈 수 있는 Enum 값이지만, 타입을 명시하지 않으면 타입스크립트는 frame 값을 `number`로 유추합니다. 그럼 코드를 작성하다 실수로 frame을 2로 설정했을 때 바로 에러가 나지 않고, UI가 깨질 수 있습니다. 이런 경우에 타입스크립트의 Union Type(합 타입)을 이용하여 상태의 범위를 한정하는 게 유용합니다!